controllers/index.js

```diff
@@ -0,0 +1,317 @@
+			if (files[key]){
+				const fileInfo = validation.getFileInfo(files[key], fileConstraints);
+				fileInfo.keyname = `${controlNumber}/${fileInfo.filename}`;
+				db.saveFile(application.id, fileInfo, function(err){
+					if (err){
+						return error.sendError(req, res, 500, `${fileInfo.filetype} failed to save`);
+					}
+					else {
+						store.uploadFile(fileInfo, callback);
+					}
+				});
+			}
+			else {
+				return callback (null, null);
+			}
+		});
+	});
+	async.parallel(asyncTasks, function(err, data){
+		if (err){
+			return callback(err, null);
+		}
+		else {
+			return callback(null, data);
+		}
+	});
+}
+
+//*******************************************************************
+// controller functions
+
+const getControlNumberFileName = function(req, res, reqData) {
+
+	const controlNumber = reqData.matches.controlNumber;
+	const fileName = reqData.matches.fileName;
+
+
+	const filePath = controlNumber + '/' + fileName;
+
+	db.getFile(filePath, function (err, file){
+
+		if (err){
+			error.sendError(req, res, 400, 'error getting file');	
+		}
+		else {
+			if (file){
+
+				store.getFile(controlNumber, fileName, function(err, data){
+
+					if (err){
+						error.sendError(req, res, 404, 'file not found');
+					}
+					else {
+						res.attachment(file.file_name);
+						res.send(data.Body);	
+					}
+
+				});
+			}
+			else {
+				error.sendError(req, res, 400, 'Invalid controlNumber or fileName provided');
+			}
+		}
+	});
+	
+};
+
+const getControlNumber = function(req, res, reqData){
+	const pathData = reqData.schema;
+	const fileTypes = {
+		'gud': 'guideDocumentation',
+		'arf': 'acknowledgementOfRiskForm',
+		'inc': 'insuranceCertificate',
+		'gse': 'goodStandingEvidence',
+		'opp': 'operatingPlan'
+	};
+
+	const basicData = getBasicRes(pathData);
+
+	let jsonData = {};
+
+	const controlNumber = reqData.matches.controlNumber;
+
+	const jsonResponse = {};
+	jsonResponse.success = true;
+	jsonResponse.api = 'FS ePermit API';
+	jsonResponse.type = 'controller';
+	jsonResponse.verb = req.method;
+	jsonResponse.src = 'json';
+	jsonResponse.route = req.originalUrl;
+
+	const cnData = basicData[1095010356];  // TODO: remove - used for mocks
+
+	if (basicData){
+		db.getApplication(controlNumber, function(err, appl, fileData){
+			if (err){
+				return error.sendError(req, res, 400, 'error getting application from database');
+			}
+			else {
+				if (fileData){
+					fileData.forEach(function(file){
+						const fileType = fileTypes[file.file_type];
+						appl[fileType] = file.file_name;
+					});
+				}
+				jsonData = get.copyGenericInfo(cnData, appl, jsonData, pathData.getTemplate);
+				jsonData.controlNumber = controlNumber;// TODO: remove - used for mocks
+				const toReturn = Object.assign({}, {response:jsonResponse}, jsonData);
+
+				res.json(toReturn);
+			}
+		});
+	}
+
+};
+
+//*************************************************************
+
+const postApplication = function(req, res, reqData){
```

Comment:
I like how this main function has gotten shorter, much easier to follow through and see the steps that are taken to post an application.